prefect_dask/utils.py

```diff
@@ -2,28 +2,79 @@
 Utils to use alongside prefect-dask.
 """
 
-from contextlib import contextmanager
+from contextlib import asynccontextmanager, contextmanager
 from datetime import timedelta
 from typing import Any, Dict, Optional, Union
 
 from distributed import Client, get_client
 from prefect.context import FlowRunContext, TaskRunContext
 
+from prefect_dask.exceptions import ImproperClientError
+
+
+def _populate_client_kwargs(
+    async_client: bool,
+    timeout: Optional[Union[int, float, str, timedelta]] = None,
+    **client_kwargs: Dict[str, Any],
+) -> Dict[str, Any]:
+    """
+    Helper method to populate keyword arguments for `distributed.Client`.
+    """
+    flow_run_context = FlowRunContext.get()
+    task_run_context = TaskRunContext.get()
+
+    if flow_run_context:
+        context = "flow"
+        task_runner = flow_run_context.task_runner
+        input_client_kwargs = task_runner.client_kwargs
+        address = task_runner._connect_to
+        asynchronous = flow_run_context.flow.isasync
+    elif task_run_context:
+        # copies functionality of worker_client(separate_thread=False)
+        # because this allows us to set asynchronous based on user's task
+        context = "task"
+        input_client_kwargs = {}
+        address = get_client().scheduler.address
+        asynchronous = task_run_context.task.isasync
+    else:
+        # this else clause allows users to debug or test
+        # without much change to code
+        context = ""
+        input_client_kwargs = {}
+        address = None
+        asynchronous = async_client
+
+    if not async_client and asynchronous:
+        raise ImproperClientError(
+            f"The {context} run is async; use `get_dask_async_client` instead"
+        )
+    elif async_client and not asynchronous:
+        raise ImproperClientError(
+            f"The {context} run is not async; use `get_dask_sync_client` instead"
+        )
+
+    input_client_kwargs["address"] = address
+    input_client_kwargs["asynchronous"] = asynchronous
+    if timeout is not None:
+        input_client_kwargs["timeout"] = timeout
+    input_client_kwargs.update(**client_kwargs)
+    return input_client_kwargs
+
 
 @contextmanager
-def get_dask_client(
+def get_dask_sync_client(
```

Comment:
I'd do `get_dask_client` for sync and `get_async_dask_client` for async. Most of our users aren't async and it'll be easier this way, I think. I'd also generally say "asynchronous dask client" and "synchronous dask client" instead of "dask sync client".